deploy/shared-components/home-index-service/Dockerfile

```diff
@@ -11,22 +11,50 @@ FROM ${BASE_IMAGE} AS builder
 # Set working directory
 WORKDIR /app
 
+# Configure Go environment with retry-friendly settings
+# Use multiple proxies with fallback, and set longer timeouts
+ENV GOPROXY=https://proxy.golang.org,direct \
+    GOSUMDB=sum.golang.org \
+    GOPRIVATE= \
+    GONOPROXY= \
+    GONOSUMDB= \
+    CGO_ENABLED=0 \
+    GOOS=linux \
+    GOARCH=amd64
+
 # Copy go.mod and go.sum from the home-index-service directory
 COPY deploy/shared-components/home-index-service/go.mod ./
 # Copy go.sum if it exists (wildcard won't fail if file doesn't exist)
 COPY deploy/shared-components/home-index-service/go.sum* ./
 
-# Download dependencies (will use cached modules from base image)
-RUN go mod download
+# Download dependencies with retry logic to handle transient network errors
```

Comment:
we can't put retry logic for every download. If there is a particular server/repository that is consistently unstable then this would make sense. But if it's just personal internet connectivity, we shouldn't add this to the code base.